### PR TITLE
Add authentication for POST requests

### LINKED ISSUE #78: Set up auth
At the very least, the POST endpoints will need authentication.

### PR DESCRIPTION
**Pull request by gravesm**
Closes #78. This provides Basic Auth for the two POST endpoints
with the username/password being provided through an environment
variable.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -35,7 +35,9 @@ def app():
 
 @pytest.fixture
 def testapp(app):
-    return TestApp(app)
+    _app = TestApp(app)
+    _app.authorization = ('Basic', ('username', 'password'))
```
Comment (JPrevost) [inline tests/conftest.py]: I'm not sure I like the default being the testapp is authorized. I may just need more :coffee:  though.

Comment (gravesm) [inline tests/conftest.py]: The alternative is to add authentication for all the POST requests. This way seemed like less work. What's the concern here? I'm open to changing things, but I'd like to understand why before I do.

**Comment by JPrevost** [comment]
Will end users actually have to use this auth, or are these protected methods expected to be used solely by scripts (i.e. is this user facing or API)?

**Comment by gravesm** [comment]
These would only be used by the automated client.

File: tests/conftest.py (hunk 1 above)
Comment (JPrevost) [inline tests/conftest.py]: My concern is that we'll forget this is here and then tests will pass without explicitly adding the auth. I'd rather a test fail until I put in Auth I think. But it's not a huge deal.

Comment (JPrevost) [inline tests/conftest.py]: Put another way, someone coming to look at this app in the future (like one of us) may look at one of these methods being tested and not immediately realize it requires authentication. If the test explicitly adds `testapp.authorization = ('Basic', ('username', 'password'))` there is less chance of not understanding what methods need auth by quickly looking at tests.

Comment (gravesm) [inline tests/conftest.py]: How about if we created an `auth_testapp` in addition to the `testapp`?

**Comment by JPrevost** [comment]
It may be worth considering API keys instead of basic auth, but for this use case I can get behind either path and this is already written ;)

File: tests/conftest.py (hunk 1 above)
Comment (JPrevost) [inline tests/conftest.py]: Ah, cool. I think that addresses my concern and also keeps the tests from turning into a mess of boilerplate. :+1:

**Comment by gravesm** [comment]
Yeah, I considered API keys, but I didn't want to make things more complicated than it needed to be for such a simple case. We may need something more flexible in the future, but we can deal with it then.

**Comment by JPrevost** [comment]
+1 to waiting for the future before overcomplicating things "just in case"

**Comment by JPrevost** [comment]
Along those same lines, I'd be fine with putting off implementing `auth_testapp` until it's more clear it is needed.

:shipit: now or after that's implemented. Your call.

**Comment by JPrevost** [comment]
:+1:

**Pull request merged**